src/editors/queryWithCopilotCodeLensProvider.ts

```diff
@@ -0,0 +1,46 @@
+import * as vscode from 'vscode';
+import EXTENSION_COMMANDS from '../commands';
+import type { SendMessageToParticipantFromInputOptions } from '../participant/participantTypes';
+import { isPlayground } from '../utils/playground';
+import { COPILOT_EXTENSION_ID } from '../participant/constants';
+
+export class QueryWithCopilotCodeLensProvider
+  implements vscode.CodeLensProvider
+{
+  constructor() {}
+
+  readonly onDidChangeCodeLenses: vscode.Event<void> =
+    vscode.extensions.onDidChange;
+
+  provideCodeLenses(document: vscode.TextDocument): vscode.CodeLens[] {
+    if (!isPlayground(document.uri)) {
+      return [];
+    }
+
+    // We can only detect whether a user has the Copilot extension active
+    // but not whether it has an active subscription.
+    const hasCopilotChatActive =
+      vscode.extensions.getExtension(COPILOT_EXTENSION_ID)?.isActive === true;
+
+    if (!hasCopilotChatActive) {
+      return [];
+    }
+
+    const options: SendMessageToParticipantFromInputOptions = {
+      prompt: 'Describe the query you would like to generate.',
```

Comment:
```suggestion
      prompt: 'Describe the query you would like to generate',
```

Let's maybe remove `.` so it looks less weird with the line that vscode adds automatically.

<img width="611" alt="Screenshot 2024-11-27 at 16 24 53" src="https://github.com/user-attachments/assets/501b0b0f-c523-4887-957a-e6fb798a7a2d">